### DISCUSSION TITLE
Welcome to sp-dev-fx-extensions Discussions!

### DISCUSSION THREAD

**Discussion by hugoabernier** [pnp/sp-dev-fx-extensions#353] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by wssro** [comment]
Hi, i'm new to SharePoint and know only a few but a work task came to me and here I am , must add a Ga script to a modern SP site. Can I have a detail deployment plan of this piece of code? Do I need to have SP admin/master rights? Thank you.

**Comment by Me1G** [comment]
Hi, I would like to use this sp-dev-fx-extensions/samples/react-command-qrcode/ to create a link to the Anyone links for a document/folder.  Can you clarify if it would allow this or does it create a link to the list item but a user would have to authenticate or sign-in to access? I would like it to just open the document without any authentication. Thanks

**Comment by teylyn** [comment]
Hello, I'm not a developer. If I would like to use one of the samples in my SharePoint site, what do I need to do? I don't have the knowledge or time to install and learn a whole development framework just to download a "hello world" SPFx app. Why is there no ready-to use download?

**Reply by sympmarc** [reply]
Hi, @teylyn! Could you explain what you're trying to accomplish specifically? I'm sure we can help you through it.

**Reply by teylyn** [reply]
Hi Marc, 

As a citizen developer I do most of my day job work in SharePoint GUI, PowerApps and Flow, and I also work a lot with Power BI. I'm  very much a low code/no code person. I speak very little JS and a few words of JQuery, but that's it. I cannot afford to spend the time on learning how to use React or the whole yo and gulp and everything involved. 

So, when I see these funky extensions on github, I would absolutely LOVE to try them out but I always wonder how I can use them and why the heck is there no installer or single download file, ready to use.

When I read the "minimal path to awesome" instruction, I recognize that the language is English, but the words don't mean anything.

Minimal Path to Awesome
Clone this repository
Ensure that you are at the solution folder
in the command-line run:
npm install
gulp serve
Include any additional steps as needed.

If this is not a big deal for the developer, maybe they could do the npm and gulp bits and make the final product available for those who want to use the sample as is?  There must be many folks like me who can handle the SharePoint GUI, but don't do the stuff in text based dialogs with black backgrounds. 

So, if there were an easy way for a Citizen Developer like me to get an SPFx extension up and running in my client's SharePoint site, that would be awesome.

**Reply by sympmarc** [reply]
Those are all very fair questions!
 
Most of these samples were built to show the possibilities for SPFx as opposed to providing production-ready solutions. That said, most of them really are production ready. But the target audience certainly has been developers, and you’re a perfect example of someone who doesn’t fit that mold but could get value from the solutions.

When we talked separately, you let  me know you're interested in this solution:
https://github.com/pnp/sp-dev-fx-extensions/tree/main/samples/react-application-news-ticker

I tried to build it for you today and didn't have much luck. There's no documentation explaining how to set it up and I've raised an issue about that.

I'll work with the team that manages this repo to improve the experience for folks like you.

cc: @hugoabernier @VesaJuvonen

**Comment by Crzyrussian09** [comment]
Hello. I have an interesting question. I was thinking of using the react-application-qna-chat or the react-application-pva-bot in the samples. I have downloaded the repository and customized the chat widget and canvas to how I want it. I was ready to upload it to my company SharePoint Intranet and use it. Then my boss asked me how do I know if there is any malicious code in the application or not... I have no idea how to answer him. I know that this repository is part of a Microsoft Community but can anyone here verify whether or not there is malicious code in those applications. I am personally not a coder and do not have much experience and wouldn't even know what to look for. If someone could take the time to answer the question, I would appreciate it. Thank you

**Reply by Crzyrussian09** [reply]
Sorry that I did not introduce my self properly first. My name is Dennis Kremenetsky. I am currently living in Finland but I am originally from America. I am soon graduating as an IT engineer and I did my internship at Marskidata OY. One of my first tasks was to figure out how to have a chat bot on a SharePoint online site. So here I am searching the web for great solutions when I came across this repository. I must say thank you to everyone who has contributed to the web parts and extensions.

**Reply by sympmarc** [reply]
@Crzyrussian09 - The PnP repos are community managed but sanctioned by Microsoft. If you go all the way down to the bottom of [the home page for the Microsoft 365 Platform Community](https://pnp.github.io/), you'll see the supportability statement. 

@hugoabernier @PopWarner @VesaJuvonen - We should have a link to the Forms, Team, and Supportability sections in the menu.
